onnxruntime/core/framework/session_options.h

```diff
@@ -79,7 +79,7 @@ struct SessionOptions {
   // enable the memory arena on CPU
   // Arena may pre-allocate memory for future usage.
   // set this option to false if you don't want it.
-  bool enable_cpu_mem_arena = true;
+  bool enable_cpu_mem_arena = false;
```

Comment:
We discussed this in the core team chat. This change will affect all customers. Have you ensured this doesn't cause any regressions?

Reply:
I actually don't expect arena will help CPU performance in most scenarios. yes it will impact all users but i think most case is positive impact? and they have the option to bring the old behavior back, right?

Reply:
An extra and unnecessary layer could cause more problems. Right now XNNPack EP's users even do not have a way to disable the arena allocator. Different allocators surely have subtle differences on performance and memory usages, but they should all work correctly in terms of the interface they aim to provide. I cannot guarantee this change will not create any trouble to any customer, but at least they have a way to migrate it.  Overall we should try to provide the best default options because we cannot think most customers know how to tune them.  And, if we disable this arena by default, it gives us a chance to introduce another allocator(e.g. mimalloc) which is more suitable for CPU EPs in default build.

Reply:
I'm not opposed to the change. The arena behavior has caused a lot of support issues already and configuring it is not easy. However, I would rather that we give users at least 1 release notice before disabling it by default. Integrating mimalloc is not the best solution either. Besides, mimalloc and the BFC arena usage in ORT are not the same thing with the former taking over all allocations and the latter only internal tensors. I would rather that we not *impose* any allocator on our users; they should be able to provide one or simply rely on other 3P solutions like mimalloc/jemalloc/tcmalloc. Mimalloc should still be available as a build option (not the default).

Reply:
agree with @pranavsharma
we should be cautious when changing default behavior which can potentially impact large % of users.
totally fine to change the behavior if the data backs it up, but we should provide sufficient notice of the change.

Reply:
Also, not sure about this statement. 

> Right now XNNPack EP's users even do not have a way to disable the arena allocator. 

Based on the code xnnpack doesn't even rely on the arena allocator and I believe that is intentional.

Reply:
But it is on the opposite side of what you were saying. This is because the constructor of AllocatorCreationInfo has many parameters, and by default arena is enabled if you do not pass-in a bool value to disable it.

Reply:
@fs-eire , this is the reason why you have a hard to reproduce bug. Now everything makes sense after I disable the arena allocator for XNNPack EP. There is nothing wrong with the arena allocator, but such an extra layer makes debugging harder.

Reply:
I will close this PR now. We can have more offline discussions before I make new attempts to fix the problem.